packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue

```diff
@@ -31,36 +48,49 @@ export default defineComponent({
     }
   },
   emits: ['createPublicLink'],
-  setup() {
+  setup(props, { emit }) {
     const { can } = useAbility()
+    const { $gettext } = useGettext()
+
     const canCreatePublicLinks = computed(() => can('create-all', 'PublicLink'))
-    return { canCreatePublicLinks }
-  },
-  computed: {
-    heading() {
-      return this.$gettext('Quick link')
-    },
-    createLinkHint() {
-      return this.$gettext('Create quick link')
-    },
-    createLinkLabel() {
-      return this.$gettext('Create link')
-    },
-    noLinkLabel() {
-      return this.$gettext('No link')
-    }
-  },
-  methods: {
-    createQuickLink() {
-      this.$emit('createPublicLink', {
+    const resource = inject<Resource>('resource')
+    const createQuickLink = () => {
+      const allowResharing = unref(useCapabilityFilesSharingResharing())
+      const canEdit = unref(useCapabilityFilesSharingPublicCanEdit())
+      const canContribute = unref(useCapabilityFilesSharingPublicCanContribute())
+      const alias = unref(useCapabilityFilesSharingPublicAlias())
+      const emitData = {
         link: {
-          name: this.$gettext('Quicklink'),
-          permissions: this.canCreatePublicLinks ? 1 : 0,
-          expiration: this.expirationDate.enforced ? this.expirationDate.default : null,
+          name: $gettext('Quicklink'),
+          permissions: LinkShareRoles.getByName(
```

Comment:
Please add the shorthand `const roleName = !canCreatePublicLink ? 'none' : capabilitiesRoleName || 'viewer'` here as well!

Reply:
sorry missed it